### DISCUSSION TITLE
How to setup DaisyUI Theme with Nextjs?

### DISCUSSION THREAD

**Discussion by myhendry** [saadeghi/daisyui#386] [Q&A] [answered]
I refer to DaisyUI documentation but it does not provide instructions on how to use DaisyUI Themes with Nextjs. Anyone can point me to materials to show me how to use DaisyUI Theme with Nextjs? Thanks

**Comment by saadeghi** [comment, marked as answer]
Here's an example: https://github.com/saadeghi/daisyui/tree/master/examples/nextjs

**Reply by myhendry** [reply]
I chance upon this repo of yours https://github.com/saadeghi/daisyui-nextjs-nextthemes. It is cool from a nextjs user

**Reply by myhendry** [reply]
I'm exploring the theme now. I notice that the Navbar component doesn't respond to the theme change. Is that suppose to be the case? or is it suppose to also respond with the theme change?

**Reply by saadeghi** [reply]
Themes should apply to the whole page. I can help finding the problem if you share your code.

**Reply by myhendry** [reply]
thanks bro. i will send you the link to my github repo shortly. i am just doing a starter page so that it is easier for me to get started on future projects. your library will help tremendously :)

**Reply by myhendry** [reply]
here goes the [repo](https://github.com/myhendry/template-web/blob/main/pages/index.tsx). the codes are all in the index.tsx page. the Header has been labelled

**Reply by saadeghi** [reply]
The navbar is using`neutral` color (by these class names: `bg-neutral text-neutral-content`) so it will be dark on both dark mode and light mode. You can remove those class names.  
Read more about colors here: https://daisyui.com/core/colors

**Reply by myhendry** [reply]
just tested. its working perfectly now! Thank you :)

**Reply by mael-bomane** [reply]
> Here's an example: https://github.com/saadeghi/daisyui/tree/master/examples/nextjs

repo answer 404...

**Reply by saadeghi** [reply]
@EsyWin All examples are moved to StackBlitz now:  
https://daisyui.com/docs/install/

Next.js example: https://stackblitz.com/edit/daisyui-nextjs

**Reply by mael-bomane** [reply]
Okay, I was searching where to throw `data-theme` since `next.js` has no `html` but throwing it into `_app.jsx` top level container does the trick :)
Sorry for notification legend !

**Reply by ChristianOConnor** [reply]
This doesn't work for me:  
```
import '../styles/globals.css'
import type { AppProps } from 'next/app'

function MyApp({ Component, pageProps }: AppProps) {
  return (
    <div theme-data="synthwave">
  <Component {...pageProps} />
  </div>
  )
}

export default MyApp
```  

Just gives the the same white default next.js page even when I have the correct tailwind.config.js
```
/** @type {import('tailwindcss').Config} */
module.exports = {
    //...
    content: ['./pages/**/*.{js,ts,jsx,tsx}'],
    plugins: [require("daisyui")],
    daisyui: {
      themes: true
    }
  }
```  

and postcss.config.js  
```
module.exports = {
    plugins: ['tailwindcss', 'autoprefixer'],
  };
```  

And I imported 
```
@tailwind base;
@tailwind components;
@tailwind utilities;
```
at the top of globals.css.

Why isn't this providing the daisyui theme?

**Reply by NayamAmarshe** [reply]
> Why isn't this providing the daisyui theme?

You're doing it wrong.

In your _app.tsx do this:

```js
import type { AppProps } from "next/app";
import "@/styles/globals.css";

function MyApp({ Component, pageProps }: AppProps<{ session: Session }>) {
  return (
    <div data-theme="cupcake">
        <Component {...pageProps} />
    </div>
  );
}

export default MyApp;
```

**Reply by ChristianOConnor** [reply]
> > Why isn't this providing the daisyui theme?
> 
> You're doing it wrong.
> 
> In your _app.tsx do this:
> 
> ```js
> import type { AppProps } from "next/app";
> import "@/styles/globals.css";
> 
> function MyApp({ Component, pageProps }: AppProps<{ session: Session }>) {
>   return (
>     <div data-theme="cupcake">
>         <Component {...pageProps} />
>     </div>
>   );
> }
> 
> export default MyApp;
> ```

Awesome that fixed it thanks!

**Reply by daveteu** [reply]
`<html>` is in _document.js of /pages/ folder.

```
import { Html, Head, Main, NextScript } from 'next/document'

export default function Document() {
  return (
    <Html data-theme="light">
      <Head />
      <body>
        <Main />
        <NextScript />
      </body>
    </Html>
  )
}
```

You can toggle dark and light theme with

```
  const toggleDarkmode = () => {
    if (darkmode == 'light') {
      document.documentElement.classList.add('dark');
      document.documentElement.setAttribute('data-theme', 'dark');
      localStorage.setItem('color-mode', 'dark');
    } else {
      document.documentElement.classList.remove('dark');
      document.documentElement.setAttribute('data-theme', 'light');
      localStorage.setItem('color-mode', 'light');
    }
  };
```

This line sets the theme to dark theme
```
      document.documentElement.setAttribute('data-theme', 'dark');
```

This line is to set darkmode for tailwind so you can use `dark:someclass`
```
      document.documentElement.classList.add('dark');
```

**Reply by irfandyj** [reply]
Hi @saadeghi, is there a new sample with the new NextJS 14?
